src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+package duke;
+import duke.tasks.*;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Duke {
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void goodbye() {
+        saveData();
+        System.out.println("\n" +
+                "░██████╗░░█████╗░░█████╗░██████╗░██████╗░██╗░░░██╗███████╗\n" +
+                "██╔════╝░██╔══██╗██╔══██╗██╔══██╗██╔══██╗╚██╗░██╔╝██╔════╝\n" +
+                "██║░░██╗░██║░░██║██║░░██║██║░░██║██████╦╝░╚████╔╝░█████╗░░\n" +
+                "██║░░╚██╗██║░░██║██║░░██║██║░░██║██╔══██╗░░╚██╔╝░░██╔══╝░░\n" +
+                "╚██████╔╝╚█████╔╝╚█████╔╝██████╔╝██████╦╝░░░██║░░░███████╗\n" +
+                "░╚═════╝░░╚════╝░░╚════╝░╚═════╝░╚═════╝░░░░╚═╝░░░╚══════╝\n" +
+                "Hope to see you again soon!");
+    }
+
+    public static void welcome() {
+        System.out.println("\n" +
+                "██╗░░██╗███████╗██╗░░░░░██╗░░░░░░█████╗░\n" +
+                "██║░░██║██╔════╝██║░░░░░██║░░░░░██╔══██╗\n" +
+                "███████║█████╗░░██║░░░░░██║░░░░░██║░░██║\n" +
+                "██╔══██║██╔══╝░░██║░░░░░██║░░░░░██║░░██║\n" +
+                "██║░░██║███████╗███████╗███████╗╚█████╔╝\n" +
+                "╚═╝░░╚═╝╚══════╝╚══════╝╚══════╝░╚════╝░\n" +
+                "from\n" + "\n" +
+                "██████╗░██╗░░░██╗██╗░░██╗███████╗\n" +
+                "██╔══██╗██║░░░██║██║░██╔╝██╔════╝\n" +
+                "██║░░██║██║░░░██║█████═╝░█████╗░░\n" +
+                "██║░░██║██║░░░██║██╔═██╗░██╔══╝░░\n" +
+                "██████╔╝╚██████╔╝██║░╚██╗███████╗\n" +
+                "╚═════╝░░╚═════╝░╚═╝░░╚═╝╚══════╝\n");
+        System.out.println("Hello! I'm Duke\nWhat can I do for you?");
+        System.out.println("_________________________________________________________________________");
+    }
+
+    public static void printInvalidMessage(String type) {
+        System.out.println("\uD83D\uDE1E OOPS! " + type + " is not valid, please try again!");
+    }
+
+    public static void validateInput (String input, String type) throws DukeException {
+        try {
+            if (type.equals("done") && (!(input.strip()).matches("[-+]?\\\\d*\\\\.?\\\\d+\n"))) {
+                throw new DukeException();
+            } else if (type.equals("todo") && (input.strip().length() <= 0)) {
+                throw new DukeException();
+            } else {
+                if (!input.contains("/")) {
+                    if (type.equals("deadline")) {
+                        throw new DukeException();
+                    } else if (type.equals("event")){
+                        throw new DukeException ();
+                    }
```

Comment:
Avoid going >3 levels of nesting. Consider making the happy path the prominent branch to reduce nesting and structure the logic around that.

https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent